services/QuillLMS/client/app/bundles/Staff/components/evidence/llmPromptDatasets/individualDataset/individualDataset.tsx

```diff
@@ -0,0 +1,121 @@
+import * as React from "react";
+import { useQuery } from 'react-query';
+import { RouteComponentProps } from 'react-router-dom';
+
+import DatasetNotesModal from './datasetNotesModal'
+import TrialsSection from './trialsSection'
+import GuidelinesSection from './guidelinesSection'
+import DatasetRelevantTextsSection from './datasetRelevantTextsSection'
+
+import DatasetTable from '../shared/datasetTable'
+import { Spinner, } from '../../../../../Shared/index';
+import { renderHeader } from '../../../../helpers/evidence/renderHelpers';
+import { getPromptIdString } from '../../../../helpers/evidence/ruleHelpers';
+import { ActivityRouteProps, DatasetInterface, StemVaultInterface, TrialInterface, } from '../../../../interfaces/evidenceInterfaces';
+import { fetchActivity } from '../../../../utils/evidence/activityAPIs';
+import { fetchDatasetAndAssociatedRecords, } from '../../../../utils/evidence/genAIAPIs';
+import { titleCase, } from '../../../../helpers/evidence/miscHelpers';
+
+const IndividualDataset: React.FC<RouteComponentProps<ActivityRouteProps>> = ({ match }) => {
+  const { params } = match;
+  const { activityId, datasetId, promptConjunction, } = params;
+
+  const [loading, setLoading] = React.useState(true)
+  const [promptIds, setPromptIds] = React.useState<string>(null);
+  const [dataset, setDataset] = React.useState<DatasetInterface>(null)
+  const [dataSubsets, setDataSubsets] = React.useState<DatasetInterface[]>(null)
+  const [stemVault, setStemVault] = React.useState<StemVaultInterface>(null)
+  const [trials, setTrials] = React.useState<TrialInterface[]>(null)
+  const [datasetRelevantTexts, setDatasetRelevantTexts] = React.useState<any>(null)
+  const [optimalGuidelines, setOptimalGuidelines] = React.useState<any>(null)
+  const [suboptimalGuidelines, setSuboptimalGuidelines] = React.useState<any>(null)
+  const [showSubsets, setShowSubsets] = React.useState<boolean>(false)
+  const [showNotesModal, setShowNotesModal] = React.useState<boolean>(false)
+
+  const { data: activityData } = useQuery({
+    queryKey: [`activity-${activityId}`, activityId],
+    queryFn: fetchActivity
+  });
+
+  const { data: datasetData } = useQuery({
+    queryKey: [`datasets-${datasetId}`, datasetId],
+    queryFn: fetchDatasetAndAssociatedRecords
+  });
+
+  React.useEffect(() => {
+    if(!promptIds && activityData && activityData.activity) {
+      const { prompts } = activityData.activity;
+      const promptIdString = getPromptIdString(prompts);
+      setPromptIds(promptIdString);
+    }
+  }, [activityData]);
+
+  React.useEffect(() => {
+    if(loading && datasetData && datasetData.data) {
```

Comment:
This is a style question: do you think it makes sense to move toward safe-accessors for things like `datasetData && datasetData.data`?  That could just be `datasetData?.data`.  I don't know that it's actually better, but the thought occurred to me as I read through this.

Reply:
Yes, you're totally right. I was following an older pattern here, and I'll change it.